### DISCUSSION TITLE
is there any way I can highlight part of the text based on given a index span within in a pdf?

### DISCUSSION THREAD

**Discussion by keyuchen21** [py-pdf/pypdf#2621] [Q&A]
is there any way I can highlight part of the text based on given a index span within in a pdf?

**Comment by stefan6419846** [comment]
Inside PDF files, text is rendered through multiple operators and with possible character mappings, thus using indices for highlighting becomes rather complex to implement. I would recommend trying to use the text visitor approach instead which yields you coordinates you should be able to use for generating annotations for the corresponding texts.

**Reply by keyuchen21** [reply]
thanksssss!
i'm wondering if you have/know any example on it?
i'm new to PDF files...

**Reply by stefan6419846** [reply]
I recommend you to have a look at our docs and get familiar with it, especially regarding text visitors at https://pypdf.readthedocs.io/en/latest/user/extract-text.html#using-a-visitor and annotations at https://pypdf.readthedocs.io/en/latest/user/adding-pdf-annotations.html#highlighting

How I would tackle this:

1. Use the text visitor to generate the correctly ordered text based upon the matrix data and save the visitor "metadata" for each word.
2. Determine the words and corresponding "metadata" you want to highlight.
3. Determine the rectangle coordinates for the highlights from the "metadata", splitting on line breaks (y coordinate changes) accordingly.
4. Use the rectangle coordinates to render the highlights.
